drivers/overlay/overlay.go

```diff
@@ -330,6 +330,14 @@ func Init(home string, options graphdriver.Options) (graphdriver.Driver, error)
 		return nil, err
 	}
 
+	supported, err := SupportsNativeOverlay(home, runhome)
+
+	if !supported {
+		if path, err := exec.LookPath("fuse-overlayfs"); err == nil && opts.mountProgram == "" {
+			opts.mountProgram = path
```

Comment:
Shouldn't this happen before the `if opts.mountProgram != "" {` block above?

Reply:
Right, it should, otherwise it could still cause issues in certain situations.